app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -389,6 +390,13 @@ class BrowserTabFragment :
 
     private val downloadMessagesJob = ConflatedJob()
 
+    /*
+      Used to prevent autofill credential picker from automatically showing
+      Useful if user has done something that would result in a strange UX to then show the picker
+      This only prevents against automatically showing; if the user taps on an autofill field directly, the dialog can still show
+     */
+    private var autofillSelectCredentialsDialogCanAutomaticallyShow = true
```

Comment:
Should this be moved to the view model so that we can also add test around it?
I have very limited context on this, so probably there's something preventing this from being part of the view state, but asking just in case

Reply:
Good question... it's already in the view state, and we currently use updates to the view state to set a local value held in the fragment. the fragment either need to keep a local var tracking it or it can instead reach out to the `ViewModel` to provide the latest value on demand.

Probably doesn't make sense to add it to the view state and then not use the view states updates for anything. so i think the two options are:

1. it's in the view state. updates result in fragment updating its local var (current code)
2. it's not in the view state. the viewmodel has a local var. the fragment queries the VM for its latest value on demand


2 marginally simpler i think. thoughts to the contrary?